tests/runtime/test_clSetEventCallback.c

```diff
@@ -145,6 +157,17 @@ int main()
 
   clFinish(queue);
 
+  i = 0;
+  while (!submit || !running || !complete)
+    {
+      sleep(1);
+      ++i;
+      if (i >= 10)
+        {
+          puts("Callback functions were not called in 10s -> assume FAIL\n");
+          return EXIT_FAILURE;
+        }
+    }
```

Comment:
If the goal was for the test to timeout, the 'timeout' command can do the same and could have stayed outside of the test without modifying it but that's a detail.